docs/building.md

```diff
@@ -0,0 +1,125 @@
+* `build/akri-containers.mk`: this provides the build and push functionality for Akri containers
+* `build/akri-rust-containers.mk`: this provides a simple definition to build and push Akri components written in Rust
+* `build/akri-dotnet-containers.mk`: this provides a simple definition to build and push Akri components written in .NET
+* `build/akri-python-containers.mk`: this provides a simple definition to build and push Akri components written in Python
+
+### Configurability
+The makefiles allow for several configurations:
+* BUILD_AMD64: if set not to 1, the make commands will ignore AMD64
+* BUILD_ARM32: if set not to 1, the make commands will ignore ARM32
+* BUILD_ARM64: if set not to 1, the make commands will ignore ARM64
+* REGISTRY: allows configuration of the container registry (defaults to imaginary: devcaptest.azurecr.io)
+* UNIQUE_ID: allows configuration of container registry account (defaults to $USER)
+* PREFIX: allows configuration of container registry path for containers
+* LABEL_PREFIX: allows configuration of container labels
+* CACHE_OPTION: allows `docker build` options to be passed into build
+
+### Local development usage
+For a local build, some typical patters are:
+* `make akri-build`: run Rust cross-build for all platforms
+* `BUILD_AMD64=0 BUILD_ARM32=0 BUILD_ARM64=1 make akri-build`: run Rust cross-build for ARM64
+* `PREFIX=ghcr.io/myaccount make akri`: builds all of the Akri containers and stores them in a container registry, `ghcr.io/myaccount`.
+* `PREFIX=ghcr.io/myaccount make akri`: builds all of the Akri containers and stores them in a container registry, `ghcr.io/myaccount`.
+* `PREFIX=ghcr.io/myaccount LABEL_PREFIX=local make akri`: builds all of the Akri containers and stores them in a container registry, `ghcr.io/myaccount` with labels prefixed with `local`.
+* `PREFIX=ghcr.io/myaccount BUILD_AMD64=1 BUILD_ARM32=0 BUILD_ARM64=0 make akri`: builds all of the Akri containers for AMD64 and stores them in a container registry, `ghcr.io/myaccount`.
+* `PREFIX=ghcr.io/myaccount make akri-controller`: builds the Akri controller container for all platforms and stores them in a container registry, `ghcr.io/myaccount`.
+
+### make targets
+For each component, there will be a common set of targets:
+* `akri-<component>`: this target will cross-build Akri and build+push this component's container for all platforms
+* `akri-docker-<component>`: this target will build+push this component's container for all platforms
+* `<component>-build`: this target will build this component's container for all platforms
+* `<component>-build-amd64`: this target will build this component's container for amd64
+* `<component>-build-arm32`: this target will build this component's container for arm32
+* `<component>-build-arm64`: this target will build this component's container for arm64
+* `<component>-docker-per-arch`: this target will push this component's container for all platforms
```

Comment:
Minor inconsistency nit: akri-docker-<component> is build+push of the container, whereas this one is only push. Wonder if we should rename to something like <component>-docker-push-per-arch.

Reply:
i agree ... but am hesitant to make the change as it will require changes to the github actions (including the intermediate build actions) and i don't really want to rerun those to validate the change.